src/go/rdctl/cmd/root.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright © 2022 SUSE LLC
+
+*/
+
+// Package cmd is the main package for this CLI
+package cmd
+
+import (
+  "encoding/json"
+  "fmt"
+  "io/ioutil"
+  "log"
+  "net/http"
+  "os"
+  "path/filepath"
+  "strconv"
+
+  "github.com/spf13/cobra"
+)
+
+var (
+  // Used for flags
+  configPath  string
+  user string
+  port string
+  password string
+)
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "rdctl",
+	Short: "A CLI for Rancher Desktop",
+	Long: `rdctl can be used to drive Rancher Desktop in headless mode.
+Supported commands include getting the current preferences,
+changing settings (with an automatic restart when needed,
+and shutting Rancher Desktop down.`,
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	err := rootCmd.Execute()
+	if err != nil {
+		os.Exit(1)
+	}
+}
+
+func init() {
+  cobra.OnInitialize(initConfig)
+
+  rootCmd.PersistentFlags().StringVar(&configPath, "config-path", "", "config file (default is $APPHOME/rancher-desktop/rd-engine.json)")
+  rootCmd.PersistentFlags().StringVar(&user, "user", "", "overrides the user setting in the config file")
+  rootCmd.PersistentFlags().StringVar(&port, "port", "", "overrides the port setting in the config file")
+  rootCmd.PersistentFlags().StringVar(&password, "password", "", "overrides the password setting in the config file")
+}
+
+func doRequest(method string, command string)  error {
+  req, err := getRequestObject(method, command)
```

Comment:
Why do we need the extra functions? What do they get us?

Reply:
I assume you mean `getRequestObject` and `doRestOfRequest`.

Thinking of the future. If/when we write an OpenAPI spec for the server, we can generate a client. And then we want an `Actor` layer to sit between this `Command` layer and the API layer. `getRequestObject` and `doRestOfRequest` correspond to two of the main functions that will be in the `Actor` layer, so it's laying groundwork for where we're going.

The two functions are also standalone and easier to reason about.

Reply:
Alright, then; I don't object to them strongly, so as long as I made you re-think it enough to explain it, it's probably good enough :)